### PR TITLE
add org.qelectrotech.QElectroTech

### PR DESCRIPTION
**Pull request by kevinsmia1939**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. https://github.com/qelectrotech/qelectrotech-source-mirror/issues/143
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.qelectrotech.QElectroTech.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {
+      "name": "qelectrotech",
+      "buildsystem": "qmake",
+      "post-install": [
+        "mv /app/share/mime/packages/qelectrotech.xml /app/share/mime/packages/org.qelectrotech.QElectroTech.xml"
+      ],
+      "sources": [
+        {
+          "type": "dir",
+          "path": "../.."
```
Comment (bilelmoussaoui) [inline org.qelectrotech.QElectroTech.json:28]: You need a tarball/git source here

File: org.qelectrotech.QElectroTech.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  "id": "org.qelectrotech.QElectroTech",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.14",
```
Comment (bilelmoussaoui) [inline org.qelectrotech.QElectroTech.json:4]: Maybe bump it to 5.15?

File: org.qelectrotech.QElectroTech.json (hunk 1 above)
Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:28]: https://git.tuxfamily.org/qet/qet.git/tag/?id=0.8.0

File: org.qelectrotech.QElectroTech.json (hunk 2 above)
Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:4]: Yes

File: org.qelectrotech.QElectroTech.json (hunk 1 above)
Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:28]: Or
https://github.com/qelectrotech/qelectrotech-source-mirror/releases/tag/0.8.0

File: org.qelectrotech.QElectroTech.json (hunk 2 above)
Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:4]: Build fine with runtime-version": "5.15" here see:
https://qelectrotech.org/forum/viewtopic.php?pid=14448#p14448

**Comment by nedrichards** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
```
$ flatpak install flathub org.kde.Platform//5.14 org.kde.Sdk//5.14
$ flatpak-builder --force-clean --ccache --user --install _build build-aux/flatpak/org.qelectrotech.QElectroTech.json
```

**Comment by barthalion** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by barthalion** [comment]
Failed to notice there are some fixes pending to make it actually buildable.

**Comment by scorpio810** [comment]
`]2;flatpak-builder: Building qelectrotech]2;flatpak-builder: Cleanup qelectrotechError: module qelectrotech: Error when getting information for file “/srv/buildbot/worker/build-x86_64-7/build/.flatpak-builder/build/vkd3d-proton-32bit-1/wine/dlls/api-ms-win-core-misc-l1-1-0”: No such file or directory`

Win build?

File: org.qelectrotech.QElectroTech.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  "rename-desktop-file": "qelectrotech.desktop",
+  "rename-appdata-file": "qelectrotech.appdata.xml",
+  "rename-icon": "qelectrotech",
+  "copy-icon": true,
+  "finish-args": [
+    "--socket=wayland",
+    "--socket=x11",
+    "--device=dri",
+    "--share=ipc",
+    "--filesystem=host"
```
Comment (tinywrkb) [inline org.qelectrotech.QElectroTech.json:16]: The `--filesystem=host` is a little too permissive for this kind of app.  
A User who blocks this permission will hit an issue with the app home folder not being kept so maybe add a comment somewhere about `~/.qet`?  or maybe just add `- --persist=.qet`?

File: org.qelectrotech.QElectroTech.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  "runtime-version": "5.14",
+  "sdk": "org.kde.Sdk",
+  "command": "qelectrotech",
+  "rename-desktop-file": "qelectrotech.desktop",
+  "rename-appdata-file": "qelectrotech.appdata.xml",
+  "rename-icon": "qelectrotech",
+  "copy-icon": true,
+  "finish-args": [
+    "--socket=wayland",
+    "--socket=x11",
```
Comment (tinywrkb) [inline org.qelectrotech.QElectroTech.json:13]: AFAIK the app works well in Wayland, so maybe change `--socket=x11` into `--socket=fallback-x11`

Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:13]: yes, you can

Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:13]: https://github.com/qelectrotech/qelectrotech-source-mirror/commit/153aecbbf311500d5eb0d340f3b9a4d58528562e

Comment (kevinsmia1939) [reply org.qelectrotech.QElectroTech.json:13]: Hi,
Do I need to edit the PR I submit or scorpio810 able to edit my PR?

**Comment by kevinsmia1939** [comment]
i mess up some thing

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

File: org.qelectrotech.QElectroTech.json (hunk 3 above)
Comment (kevinsmia1939) [reply org.qelectrotech.QElectroTech.json:16]: Need input from @scorpio810 on this.

**Comment by kevinsmia1939** [comment]
I don't understand why the patch failed, it build just fine locally on my machine.

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by tinywrkb** [comment]
Switch to git sources, there are a few git calls during build time, see `$ grep 'shell git' -R ./`.  
Use Github mirror for fetching source, `git.tuxfamily.org` seems to be a bit slow and not supporting shallow clone.

```
diff --git a/org.qelectrotech.QElectroTech.json b/org.qelectrotech.QElectroTech.json
index e1df498..cbdfedf 100644
--- a/org.qelectrotech.QElectroTech.json
+++ b/org.qelectrotech.QElectroTech.json
@@ -24,9 +24,10 @@
       ],
       "sources": [
         {
-          "type": "archive",
-          "url": "https://git.tuxfamily.org/qet/qet.git/snapshot/qet-0.8.0.tar.gz",
-          "sha256": "62ea5bd34d823c2d287fa58db8b7452825d7c3479b8ceb6650ef630c3976e070"
+          "type": "git",
+          "url": "https://github.com/qelectrotech/qelectrotech-source-mirror.git",
+          "tag": "0.8.0",
+          "commit": "664537c1787d5ca4d5420bb369668d67e9d72d84"
         },
         {
           "type": "patch",
```

You need to fix `fix_appdata_file.patch`. This is just a quick edit by hand.
```
diff --git a/fix_appdata_file.patch b/fix_appdata_file.patch
index 8347dd3..c893d24 100644
--- a/fix_appdata_file.patch
+++ b/fix_appdata_file.patch
@@ -14,13 +14,14 @@ diff -ruN a/misc/qelectrotech.appdata.xml b/misc/qelectrotech.appdata.xml
   <name>QElectroTech</name>
   <summary>Electrical diagram editor</summary>
   <summary xml:lang="ar">مُحرّر مخططات كهربائية</summary>
-@@ -21,6 +21,9 @@
+@@ -21,6 +21,10 @@
   <summary xml:lang="be">Elektrische schema editor</summary>
   <summary xml:lang="da">Elektrisk diagram redigering</summary>
   <summary xml:lang="ja">電気回路図エディタ</summary>
 + <content_rating type="oars-1.1"/>
 + <releases>
-+     <release version="0.8.0" date="2021-02-21">
++     <release version="0.8.0" date="2021-02-21"/>
++ </releases>
   <description>
    <p>
     QElectroTech is a Qt5 application to design electric diagrams.
```

File: fix_appdata_file.patch [hunk 5]
```diff
@@ -20,7 +20,7 @@ diff -ruN a/misc/qelectrotech.appdata.xml b/misc/qelectrotech.appdata.xml
   <summary xml:lang="ja">電気回路図エディタ</summary>
 + <content_rating type="oars-1.1"/>
 + <releases>
-+     <release version="0.8.0" date="2021-02-21">
++     <release version="0.8.0" date="2021-02-21"/>
```
Comment (tinywrkb) [inline fix_appdata_file.patch:23]: You're still missing the `</releases>` end tag.

Comment (bilelmoussaoui) [reply fix_appdata_file.patch:23]: No, this is xml and it's correct.

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by tinywrkb** [comment]
> tag-invalid           : <metadata_license> is not valid [GPL-2.0+]

[Metadata Quickstart: 4.1.3 Recommended metadata file contents#<metadata_license/>](https://freedesktop.org/software/appstream/docs/chap-Quickstart.html#qsr-app-contents)

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
Maybe add --socket=cups

see https://qelectrotech.org/forum/viewtopic.php?pid=14700#p14700

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
Great.
I don't know if this issue if fixed https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/4569
It would be nice to add if possible qet_tb_generator plugin in same side.
https://pypi.org/project/qet-tb-generator/

https://download.tuxfamily.org/qet/scorpio/org.qelectrotech.QElectroTech.json

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by kevinsmia1939** [comment]
@scorpio810 Not sure if this work, please take a look.

**Comment by scorpio810** [comment]
@kevinsmia1939 for check if work, run qelectrotech in terminal open industrial.qet example, now go to project menu and click on launch qet_tb_generator plugin
See: https://qelectrotech.org/forum/viewtopic.php?pid=11294#p11294

For check dxftoqet open qelectrotech element editor in menu file click on open starting the DXF converter plugin.

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by kevinsmia1939** [comment]
> @kevinsmia1939 for check if work, run qelectrotech in terminal open industrial.qet example, now go to project menu and click on launch qet_tb_generator plugin
> See: https://qelectrotech.org/forum/viewtopic.php?pid=11294#p11294
> 
> For check dxftoqet open qelectrotech element editor in menu file click on open starting the DXF converter plugin.

I got qet_tb_generator to work. But when I try DXF converter plugin, nothing happen, not even in terminal log.

**Comment by scorpio810** [comment]
> > @kevinsmia1939 for check if work, run qelectrotech in terminal open industrial.qet example, now go to project menu and click on launch qet_tb_generator plugin
> > See: https://qelectrotech.org/forum/viewtopic.php?pid=11294#p11294
> > For check dxftoqet open qelectrotech element editor in menu file click on open starting the DXF converter plugin.
> 
> I got qet_tb_generator to work. But when I try DXF converter plugin, nothing happen, not even in terminal log.

Great for qet_tb_generator. ;-)
For DXF converter see code here https://git.tuxfamily.org/qet/qet.git/tree/sources/editor/ui/qetelementeditor.cpp#n941
It's work by menu in element editor if is installed on user qet directory, I don't know if is possible on flatpak?

**Comment by scorpio810** [comment]
Latest qet_tb_generator version is here
https://pypi.org/project/qet-tb-generator/#files

https://files.pythonhosted.org/packages/9e/c3/aaad3309a5f2ca08e2fa0ab01123bb6fafb15a92854bff88042039192a67/qet_tb_generator-1.1.7.tar.gz

sha256
f626ab7bf4d9091fc85f63d33136585e611272b347499b448ece4e33cc04eeed

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

File: org.qelectrotech.QElectroTech.json (hunk 3 above)
Comment (nedrichards) [reply org.qelectrotech.QElectroTech.json:16]: Yes, what do you need full host permissions for? As @tinywrkb says we should allow the absolute minimum surface that lets the app access its configuration (ideally that uses xdg-config dirs so it doesn't need a permission at all, but a --persist=.qet could work too - and then as a modern Qt app I'd hope that it should be able to use the file portal to open and save files anywhere on the system as required without extra permissions.

You can test this yourself locally with --nofilesystem=host and the persist flag mentioned above on the flatpak CLI - or use an app like flatseal https://flathub.org/apps/details/com.github.tchx84.Flatseal to experiment with the permissions and minimise them.

Certainly host permissions require explicit justification that is core to the purpose of the app.

Comment (barthalion) [reply org.qelectrotech.QElectroTech.json:16]: I think qelectrotech doesn't use portals. It's not possible to save a project if you take away the host permission.

**Comment by scorpio810** [comment]
> > @kevinsmia1939 for check if work, run qelectrotech in terminal open industrial.qet example, now go to project menu and click on launch qet_tb_generator plugin
> > See: https://qelectrotech.org/forum/viewtopic.php?pid=11294#p11294
> > For check dxftoqet open qelectrotech element editor in menu file click on open starting the DXF converter plugin.
> 
> I got qet_tb_generator to work. But when I try DXF converter plugin, nothing happen, not even in terminal log.

Maybe add dxf to bin like snap?

```
 dxf-to-qet:
    after: [kde-sdk-setup]
    plugin: nil
    source: https://github.com/qelectrotech/DXFtoQET-2020.git
    override-build: |
      qmake "$SNAPCRAFT_PART_SRC/DXFtoQET.pro"
      make -j$(nproc)
      mkdir -p "$SNAPCRAFT_PART_INSTALL/bin"
      cp DXFtoQET "$SNAPCRAFT_PART_INSTALL/bin/"
```

**Comment by scorpio810** [comment]
For the DXF converter, the user can add the converter binary directly to his ~ / .qet directory and it should work.
So maybe it is not useful to put it in the flatpak if it is too complicated?

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
Pease update qet_tb_version

```
 "sources": [
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/9e/c3/aaad3309a5f2ca08e2fa0ab01123bb6fafb15a92854bff88042039192a67/qet_tb_generator-1.1.7.tar.gz",
            "sha256": "f626ab7bf4d9091fc85f63d33136585e611272b347499b448ece4e33cc04eeed"
        }
    ]
},
```

**Comment by scorpio810** [comment]
DXF starts but cannot be used.

```
flatpak run --command=DXFtoQET org.qelectrotech.QElectroTech 
Qt: Session management error: Could not open network socket
QMetaObject::connectSlotsByName: No matching signal for on_progressBar_valueChanged(int)
QMetaObject::connectSlotsByName: No matching signal for on_progressBar_valueMin(int)
QMetaObject::connectSlotsByName: No matching signal for on_progressBar_valueMax(int)
QMetaObject::connectSlotsByName: No matching signal for on_progressBar_text(QString)
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected" 
 "Not connected to D-Bus server"
QSqlDatabasePrivate::removeDatabase: connection 'qt_sql_default_connection' is still in use, all queries will cease to work.
QSqlDatabasePrivate::addDatabase: duplicate connection name 'qt_sql_default_connection', old connection removed.
QSqlDatabasePrivate::removeDatabase: connection 'qt_sql_default_connection' is still in use, all queries will cease to work.
QSqlDatabasePrivate::addDatabase: duplicate connection name 'qt_sql_default_connection', old connection removed.
```

Maybe we need to remove DXFtoQET in org.qelectrotech.QElectroTech.json?

**Comment by kevinsmia1939** [comment]
@scorpio810 
DXFtoQET does not work because it is not in user directory?

**Comment by scorpio810** [comment]
Maybe, but DXFtoQET is a standalone program.
If I start by command "flatpak run --command=DXFtoQET org.qelectrotech.QElectroTech "in setting I can't chose in dialog box where save elements converted.
2) when click on "create QET elmt" I saw error on sql 

Flatpak users can get DXFtoQET here https://download.tuxfamily.org/qet/builds/dxf_to_elmt/linux/
Put this binary on your ~/.qet directory make it executable by chmod + ./DXFtoQET

To check see 

https://qelectrotech.org/forum/viewtopic.php?pid=14765#p14765

File: org.qelectrotech.QElectroTech.json [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+          "path": "fix_appdata_file.patch"
+        },
+        {
+          "type": "patch",
+          "path": "0001-build-Fix-the-installation-paths.patch"
+        }
+      ]
+    },
+    {
+    "name": "python3.6.0",
```
Comment (barthalion) [inline org.qelectrotech.QElectroTech.json:87]: Do you really need this specific version? Is there anything wrong with the one shipped in the runtime? If you need older, why the latest 3.6.x?

**Comment by barthalion** [comment]
Yeah, DXFtoQET's behavior from flatpak is different from the one you are providing, although in both cases I failed to convert some random DXF file from the Internet. If it's not critical to have, I'd add an x86_64-only module and download a binary release instead.

File: org.qelectrotech.QElectroTech.json (hunk 6 above)
Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:87]: Requires python 3.5 or above.

Comment (barthalion) [reply org.qelectrotech.QElectroTech.json:87]: The runtime ships 3.8.8. Would it be okay?

Comment (scorpio810) [reply org.qelectrotech.QElectroTech.json:87]: Yes.

**Comment by scorpio810** [comment]
It's not critical to don't have DXF converter in Flatpak, users can get binary or compile sources.
It is a tool which is seldom used, and again it is often necessary to rework and clean the DXF ASCII  file with Inkscape to hope for a good conversion. 
https://qelectrotech.org/forum/viewtopic.php?pid=10705#p10705

**Comment by barthalion** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by barthalion** [comment]
@kevinsmia1939 Looks like your latest commits overwrote the manifest with a different app.

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by kevinsmia1939** [comment]
I remove python 3.6 and DFXtoQET for now. How about we ship it without DFXtoQET first?

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
> I remove python 3.6 and DFXtoQET for now. How about we ship it without DFXtoQET first?

Yes, no problem.
Thanks

**Comment by kevinsmia1939** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
@kevinsmia1939 Raul provide new version on qet_tb_generator 1.2.5, but this version need PySimpleGUI depend.

https://pypi.org/project/qet-tb-generator/#copy-hash-modal-cbf95192-6fb4-48f0-aa6f-9e4441c95565

See :
https://qelectrotech.org/forum/viewtopic.php?pid=14954#p14954

**Comment by kevinsmia1939** [comment]
@scorpio810 
This one? https://pypi.org/project/PySimpleGUI/

**Comment by scorpio810** [comment]
Tried https://github.com/flathub/flathub/pull/2146/commits/79b092b13d77a64d588d712f1c8c41a7f7fecadd on 0.9-dev, work great.
Thanks kevinsmia1939.

**Comment by kevinsmia1939** [comment]
Ok, I think it is ready now.

**Comment by barthalion** [comment]
bot, build org.qelectrotech.QElectroTech

**Comment by scorpio810** [comment]
bot, build org.qelectrotech.QElectroTech

**Pull request closed without merging**